hazelcast/src/main/java/com/hazelcast/internal/partition/impl/InternalPartitionServiceImpl.java

```diff
@@ -666,6 +659,11 @@ private void finalizeOrRollbackMigration(PartitionRuntimeState partitionState) {
         }
 
         updateAllPartitions(partitionTable);
+        partitionStateManager.setVersion(partitionState.getVersion());
+        if (!partitionStateManager.setInitialized()) {
```

Comment:
`partitionStateManager` may be uninitialized at this point on non-master nodes. That means this is the initial partition assignment.